File: QuickFiler/QfcViewer.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Windows.Forms;

namespace QuickFiler
{

    public partial class QfcViewer
    {
        private bool _tipsActive = true;
        private readonly List<Label> _tipsLabels = new List<Label>();
        private readonly Dictionary<ColumnStyle, float> _tipsColumns = new Dictionary<ColumnStyle, float>();

        public QfcViewer()
        {

            // This call is required by the designer.
            InitializeComponent();

            // Add any initialization after the InitializeComponent() call.
            InitTipsLabels();
            InitTipsColumns();
            ToggleAccelerator();

        }

        private void InitTipsLabels()
        {
            {
                ref var withBlock = ref this._tipsLabels;
                withBlock.Add(LblAcOpen);
                withBlock.Add(LblAcPopOut);
                withBlock.Add(LblAcTask);
                withBlock.Add(LblAcDelete);
                withBlock.Add(LblAcAttachments);
                withBlock.Add(LblAcConversation);
                withBlock.Add(LblAcEmail);
                withBlock.Add(LblAcFolder);
                withBlock.Add(LblAcSearch);
            }
        }

        private void InitTipsColumns()
        {
            {
                ref var withBlock = ref this._tipsColumns;
                withBlock.Add(L1h1L2v.ColumnStyles[0], 50f);
                withBlock.Add(L1h2L2v1h.ColumnStyles[1], 20f);
                withBlock.Add(L1h2L2v2h.ColumnStyles[1], 20f);
                withBlock.Add(L1h2L2v3h.ColumnStyles[1], 20f);
                withBlock.Add(L1h2L2v3h.ColumnStyles[5], 20f);
                withBlock.Add(L1h2L2v3h.ColumnStyles[7], 20f);
            }
        }

        private void ControlGroup_Paint(object sender, PaintEventArgs e)
        {

            if (BorderStyle == BorderStyle.FixedSingle)
            {
                int thickness = 2;
                int halfThickness = (int)Math.Round(thickness / 2d);

                using (var p = new Pen(Color.Black, thickness))
                {
                    e.Graphics.DrawRectangle(p, new Rectangle(halfThickness, halfThickness, ClientSize.Width - thickness, ClientSize.Height - thickness));
                }
            }
        }

        public void ToggleAccelerator()
        {
            if (_tipsActive)
            {
                // Make tips invisible
                foreach (Label tip in _tipsLabels)
                    tip.Visible = false;

                LblPos.Visible = false;

                // Make tips columns 0 pixels in width
                foreach (ColumnStyle col in _tipsColumns.Keys)
                    col.Width = 0f;
                _tipsActive = false;
            }
            else
            {
                // Make tips visible
                foreach (Label tip in _tipsLabels)
                    tip.Visible = true;
                LblPos.Visible = true;

                // Make tips columns 20 pixels in width
                foreach (ColumnStyle col in _tipsColumns.Keys)
                    col.Width = _tipsColumns[col];
                _tipsActive = true;
            }
        }
    }
}

File: QuickFiler/QfcViewer.vb
Imports System.Drawing
Imports System.Windows.Forms

Public Class QfcViewer
    Private _tipsActive As Boolean = True
    Private ReadOnly _tipsLabels As List(Of Label) = New List(Of Label)
    Private ReadOnly _tipsColumns As Dictionary(Of ColumnStyle, Single) = New Dictionary(Of ColumnStyle, Single)

    Public Sub New()

        ' This call is required by the designer.
        InitializeComponent()

        ' Add any initialization after the InitializeComponent() call.
        InitTipsLabels()
        InitTipsColumns()
        ToggleAccelerator()

    End Sub

    Private Sub InitTipsLabels()
        With _tipsLabels
            .Add(Me.LblAcOpen)
            .Add(Me.LblAcPopOut)
            .Add(Me.LblAcTask)
            .Add(Me.LblAcDelete)
            .Add(Me.LblAcAttachments)
            .Add(Me.LblAcConversation)
            .Add(Me.LblAcEmail)
            .Add(Me.LblAcFolder)
            .Add(Me.LblAcSearch)
        End With
    End Sub

    Private Sub InitTipsColumns()
        With _tipsColumns
            .Add(Me.L1h1L2v.ColumnStyles.Item(0), 50)
            .Add(Me.L1h2L2v1h.ColumnStyles.Item(1), 20)
            .Add(Me.L1h2L2v2h.ColumnStyles.Item(1), 20)
            .Add(Me.L1h2L2v3h.ColumnStyles.Item(1), 20)
            .Add(Me.L1h2L2v3h.ColumnStyles.Item(5), 20)
            .Add(Me.L1h2L2v3h.ColumnStyles.Item(7), 20)
        End With
    End Sub

    Private Sub ControlGroup_Paint(sender As Object, e As PaintEventArgs) Handles Me.Paint

        If Me.BorderStyle = BorderStyle.FixedSingle Then
            Dim thickness As Integer = 2
            Dim halfThickness As Integer = thickness / 2

            Using p As Pen = New Pen(Color.Black, thickness)
                e.Graphics.DrawRectangle(p, New Rectangle(halfThickness, halfThickness, Me.ClientSize.Width - thickness, Me.ClientSize.Height - thickness))
            End Using
        End If
    End Sub

    Public Sub ToggleAccelerator()
        If _tipsActive Then
            'Make tips invisible
            For Each tip As Label In _tipsLabels
                tip.Visible = False
            Next

            Me.LblPos.Visible = False

            'Make tips columns 0 pixels in width
            For Each col As ColumnStyle In _tipsColumns.Keys
                col.Width = 0
            Next
            _tipsActive = False
        Else
            'Make tips visible
            For Each tip As Label In _tipsLabels
                tip.Visible = True
            Next
            Me.LblPos.Visible = True

            'Make tips columns 20 pixels in width
            For Each col As ColumnStyle In _tipsColumns.Keys
                col.Width = _tipsColumns(col)
            Next
            _tipsActive = True
        End If
    End Sub
End Class
